test/functional/sqlsrv/sqlsrv_ae_insert_sqltype_datetime_precision.phpt

```diff
@@ -0,0 +1,181 @@
+--TEST--
+Test for inserting encrypted data of datetime2, datetimeoffset and time datatypes with different precisions
+--DESCRIPTION--
+Test implicit conversions between different precisions
+With Always Encrypted, implicit conversion works if:
+1. From input of SQLSRV_SQLTYPE_DATETIME2 to a dateteim2(7) column
+2. From input of SQLSRV_SQLTYPE_DATETIMEOFFSET to a datetimeoffset(7) column
+3. From input of SQLSRV_SQLTYPE_TIME to a time(7) column
+Note: with AE, implicit converion should work as long as the SQLSRV_SQLTYPE has a smaller precision than the one defined in the column. However, the SQLSRV driver does not let the user specify the precision in these SQLSRV_SQLTYPE_* constants and they are all default to a precision of 7. Hence when user specifies SQLSRV_SQLTYPE_DATETIME2, SQLSRV_SQLTYPE_DATETIMEOFFSET or SQLSRV_SQLTYPE_TIME when binding parameter during insertion, only insertion into a column of precision 7 is allowed.
+Without AlwaysEncrypted, implicit conversion between different precisions works
+--SKIPIF--
```

Comment:
Be consistent in your comment, either AE or Always Encrypted :)

Reply:
OK I'll change all to Always Encrypted